R/render-plot.R

```diff
@@ -570,6 +570,17 @@ find_panel_info_api <- function(b) {
       domain$bottom <- -domain$bottom
     }
 
+    # Remember the x/y range if it's a faceted
+    # plot with a free discrete axis. This is necessary
+    # to properly inverse map the numeric (i.e., trained)
+    # positions back to the data scale (#2410)
+    if (nrow(layout) > 1) {
+      free_x <- isTRUE(b$layout$facet$params$free$x)
+      free_y <- isTRUE(b$layout$facet$params$free$y)
+      if (free_x && xscale$is_discrete()) domain$xrange <- xscale$range$range
```

Comment:
`xrange` seems a little confusing, since we use `range` to mean something else in other places. Maybe `x_target` or `x_discrete_levels`?

It would also help to have an example of the data structures that come out of this.

Reply:
New thought: inside of  `domain`, it could look like:

```
#   .. ..$ domain    :List of 4
#   .. .. ..$ left  : num 1.32
#   .. .. ..$ right : num 5.62
#   .. .. ..$ bottom: num 9.22
#   .. .. ..$ top   : num 35.1
#           $ discrete_mapping
#              $ x: chr "level1" "level2" "level3"
#              $ y: chr "something1" "something2"
```